subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformValuesInjectionIntegrationTest.groovy

```diff
@@ -87,17 +87,23 @@ class ArtifactTransformValuesInjectionIntegrationTest extends AbstractDependency
 """
 
         when:
+        if (expectedDeprecation) {
+            executer.expectDeprecationWarning()
+        }
         run(":a:resolve")
 
         then:
         outputContains("processing b.jar")
         outputContains("processing c.jar")
         outputContains("result = [b.jar.green, c.jar.green]")
+        if (expectedDeprecation) {
+            outputContains(expectedDeprecation)
+        }
```

Comment:
I've been planning for a long time to actually require the deprecation message to be part of the `expectDeprecationWarning()` call...